[IMP] l10n: update CodaBox documentation

RicGR98 opened the pull request:
This commit updates the documentation of CodaBox:
- new connection process
- new revocation process
- modified potential errors

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/7683).

RicGR98 commented:
Hello @chiaraprattico 
Could I have a review for this PR ?
Thanks ! :D

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,37 @@ Configure the Journals
-   .. tab:: Following connection
+      .. note::
+        This :guilabel:`Fiduciary Password` will be used to create new connections
+        for other clients without having to validate the connection again. This password
```
for other clients without validating the connection again. This password

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,37 @@ Configure the Journals
-      :guilabel:`Connected`.
+      .. image:: belgium/codabox_connection_password.png
+        :align: center
+        :alt: Password request wizard
+#. The :guilabel:`Status` should have now switched to :guilabel:`Connected`.
```
there should be a blank line between picture and line 327

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,37 @@ Configure the Journals
-          By having a different :guilabel:`Access Token` for each client, you can
-          easily revoke the access of a client to your CodaBox account without
-          affecting the other clients. One client cannot interfere with another.
+    .. image:: belgium/codabox_connection_ok.png
+       :align: center
+       :alt: Connection status OK
```
I don't think this is necessary. We tend to avoid having too many pictures for maintenance purposes https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#images

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
-It seems that the fiduciary VAT number you provided is not valid. Please check your configuration.
-  Either the Company VAT of the Accounting Firm VAT is in the wrong format.
+No connection exists with these fiduciary and company VAT numbers. Please check your configuration.
+  This can happen when checking the connection status, and the pair (Fiduciary VAT, Company VAT)
+  is not registered yet. This may happen if you have changed the company VAT after the connection
```
This can happen when checking the connection status, and the pair (Fiduciary VAT, Company VAT) still needs to be registered.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
 
-It seems that the fiduciary VAT number you provided is not valid. Please check your configuration.
-  Either the Company VAT of the Accounting Firm VAT is in the wrong format.
+No connection exists with these fiduciary and company VAT numbers. Please check your configuration.
```
No connection exists between some accounting firms and company VAT numbers. Please check your configuration.

Not sure it should be 'these', did you mention some specific ones before, I might have missed it?

RicGR98 replied:
these refers to the ones displayed on the UI. Perhaps I should rephrase that ?

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  was established. For security reasons, you have to recreate a connection for this company VAT.
+
+It seems that your CodaBox connection is not valid anymore. Please connect again.
+  This can happen if you have revoked the access of Odoo to your CodaBox account, or that you have
+  not completed the configuration process. In this case, you must revoke your connection and create
```
This can happen if you revoke Odoo access to your CodaBox account or still need to complete the configuration process.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  not completed the configuration process. In this case, you must revoke your connection and create
+  a new one.
+
+It seems that you have already created an Odoo-CodaBox connection with this fiduciary. You must reuse the password you received from Odoo during your first connection.
```
You have already created an Odoo-CodaBox connection with this accounting firm. You must reuse the password you received from Odoo during your first connection.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  Odoo connection on CodaBox's side (i.e. on your myCodaBox portal). Then, you will be able to
+  revoke the connection on Odoo's side and create a new one.
+
+The provided password is not valid for this fiduciary.
```
The provided password is not valid for this accounting firm.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  revoke the connection on Odoo's side and create a new one.
+
+The provided password is not valid for this fiduciary.
+    The password you provided is not the one you received from Odoo during your first connection.
```
The password you provided is different from the one you received from Odoo during your first connection.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+The provided password is not valid for this fiduciary.
+    The password you provided is not the one you received from Odoo during your first connection.
+    You must use the password you received from Odoo during your first connection to create a new
+    connection for this fiduciary. If you have lost your password, you will have to first revoke the
```
connection for this accounting firm.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+    Odoo connection on CodaBox's side (i.e. on your myCodaBox portal). Then, you will be able to
+    revoke the connection on Odoo's side and create a new one.
+
+It seems that the company or fiduciary VAT number you provided is not valid. Please check your configuration.
```
The company or accounting firm VAT number you provided is not valid. Please check the configuration.

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +366,49 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
-  This may occur if the company VAT has changed after the connection was established.
-  In that case, you must click on the :guilabel:`Revoke connection` button in the settings,
-  and recreate the connection.
+  The Fiduciary VAT number you provided is not registered in CodaBox. You may not have a valid
```
The Accounting Firm VAT number you provided is not registered in CodaBox. You may not have a valid

chiaraprattico reviewed (changes requested):
Hi @RicGR98, let me know if my comments are clear.
I'd avoid use of screenshots if it's clear without one, the page is becoming extremely long and the maintenance can be more challenges with more pictures.
Here are some general info on images, we try not to overload our pages unless it really makes a difference to improve the users' understanding. https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#images
Please go to the line at the 100th character.
I know on Odoo is marked on fiduciary but this is not an English term, it is literally translated from french, please refer instead to accounting firms

RicGR98 commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,37 @@ Configure the Journals
-:menuselection:`Settings --> Users & Companies --> Groups`, and search for the group
-:guilabel:`Allow to show and modify the CodaBox Connection Settings`. Add the users you want
-to this group. By default, only the :guilabel:`Administrator` is part of this group.
+#. Go to :menuselection:`Accounting --> Configuration --> Accounting: CodaBox`.
```
It's in the Accounting section of the Configuration
![image](https://github.com/odoo/documentation/assets/32981518/51eea1fc-2341-49d0-bc04-366485f8449c)

chiaraprattico replied:
I see then I'd say :menuselection:`Accounting --> Configuration` then go to the :guilabel:`CodaBox' section

RicGR98 commented:
Thanks for the review :)

>  I'd avoid use of screenshots 

I've removed some of them, and also reworked the Configure Connection section, could you please recheck it ?

> Please go to the line at the 100th character

This is done, the only lines > 100 chars are the ones in potential issues for the same reason as https://github.com/odoo/documentation/pull/6425. I'll put the reason again in this PR description

> I know on Odoo is marked on fiduciary but this is not an English term, it is literally translated from french, please refer instead to accounting firms

Indeed, I used the term that was used in the UI to be sure the user could find it. I've renamed all code usages to Accounting Firm instead of Fiduciary :)

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
 
-Then, to configure the connection itself, go to
-:menuselection:`Accounting --> Configuration --> Accounting: CodaBox`.
+   If this is **not your first connection**, the :guilabel:`Accounting Firm Password` provided by
```
sorry here Fiduciary was correct cause it's what's written on the software

chiaraprattico commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
-       #. Copy the :guilabel:`Access Token` from a previous connection.
-       #. Click on :guilabel:`Connect`.
+      .. note::
+        This :guilabel:`Accounting Firm Password` is unique to Odoo and must be stored securely
```
same here when using guilabel it should be what's on the software exactly

chiaraprattico reviewed (commented):
Hi @RicGR98 , only left a few comments but it's looking great

RicGR98 commented:
Actually I have modified the Odoo code to use Accounting Firm instead of Fiduciary :)

![image](https://github.com/odoo/documentation/assets/32981518/300276c2-0bb4-4ea9-b929-ac7c97db0580)

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
-          By having a different :guilabel:`Access Token` for each client, you can
-          easily revoke the access of a client to your CodaBox account without
-          affecting the other clients. One client cannot interfere with another.
+#. The :guilabel:`Status` should have now switched to :guilabel:`Connected`.
```
Since this is not an actual step to be performed by the user, I would take it out of the list.

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+    :menuselection:`Accounting --> Configuration --> Accounting: CodaBox --> Manage Connection`
+    and clicking on the :guilabel:`Revoke` button.
 
 Potential issues
```
- I think the best solution for this section would be to use **bold** for the first line (the sentence(s) stating the potential issue), and maybe even a list (not sure about this one, up to you). That way you'd still be able to comply with the 100-char limit while keeping the section neat and clear. Don't forget to change the alignment of the "solution" lines :slightly_smiling_face: 
- I would also use guilabel where you're referring to the actual field labels (namely, Accounting Firm VAT and Company VAT), or take out the capital letters altogether if you're not.
- This is just a suggestion, but what about adding refs in the solution paragraphs so that people could click on the link and be taken directly to where the steps to perform are described? I don't think it's necessary for every solution, but it could be useful e.g., where you talk about revoking access or creating a new connection. What do you think?

RicGR98 replied:
Done, it's indeed cleaner :)

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
-  Then, you will have to follow the steps in the
-  :ref:`first connection <belgium/codabox-configuration-connection>` section.
+No connection exists with these accounting firms and company VAT numbers. Please check your configuration.
+  This can happen when checking the connection status, and the pair
+  (Accounting Firm VAT, Company VAT) still needs to be registered. This may happen if you have
```
```suggestion
  This can happen when checking the connection status, and the Accounting Firm VAT and Company VAT combination still needs to be registered. This may happen if you have
```

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  The password you provided is different from the one you received from Odoo during your first
+  connection.
+  You must use the password you received from Odoo during your first connection to create a new
+  connection for this accounting firm. If you have lost your password, you will have to first revoke
+  the Odoo connection on CodaBox's side (i.e. on your myCodaBox portal). Then, you will be able to
+  revoke the connection on Odoo's side and create a new one.
```
```suggestion
  connection for this accounting firm. If you have lost your password, you must first revoke
  the Odoo connection on CodaBox's side (i.e., on your myCodaBox portal). Then, you can
  revoke the connection on Odoo's side and create a new one.
```

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  revoke the connection on Odoo's side and create a new one.
+
+It seems that the company or accounting firm VAT number you provided is not valid. Please check your configuration.
+  Either the Company VAT of the Accounting Firm VAT is not in a valid Belgian format.
```
```suggestion
  Either the Company VAT or the Accounting Firm VAT is not in a valid Belgian format.
```

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
+  CodaBox license linked to this VAT number.
+
+It seems you have already created a connection to CodaBox with this accounting firm. To create a new connection, you must first revoke the old one on myCodaBox portal.
+  You must go on your myCodaBox portal and revoke the access of Odoo to your CodaBox account.
+  Then, you will be able to create a new connection on Odoo's side.
```
```suggestion
  You must go to your myCodaBox portal and revoke Odoo's access to your CodaBox account.
  Then, you can create a new connection on Odoo's side.
```

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
-:guilabel:`Allow to show and modify the CodaBox Connection Settings`. Add the users you want
-to this group. By default, only the :guilabel:`Administrator` is part of this group.
+#. Go to :menuselection:`Accounting --> Configuration --> Accounting: CodaBox`.
+#. Click on :guilabel:`Manage Connection` to open the connection wizard which shows the
```
```suggestion
#. Click on :guilabel:`Manage Connection` to open the connection wizard, which shows the
```

auva-odoo commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
+   for the connection.
+#. If this is your **first connection**, you can simply click on the :guilabel:`Create connection`.
+   The wizard confirms that the connection has been created on **Odoo's side**. Follow the steps
+   to validate the connection on **CodaBox's side** too.
```
```suggestion
   to validate the connection on **CodaBox's side**, too.
```

RicGR98 commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -377,70 +352,45 @@ Once the connection is established, Odoo can be synchronized with CodaBox.
-    :ref:`first connection <belgium/codabox-configuration-connection>` section.
+    The connection between Odoo and CodaBox can be revoked by going to
+    :menuselection:`Accounting --> Configuration --> Accounting: CodaBox --> Manage Connection`
+    and clicking on the :guilabel:`Revoke` button.
```
I feel like a tip would be something that we advised which we don't want to here IMO, what do you think ?

RicGR98 commented on content/applications/finance/fiscal_localizations/belgium.rst:
```diff
@@ -298,48 +298,23 @@ Configure the Journals
-:menuselection:`Settings --> Users & Companies --> Groups`, and search for the group
-:guilabel:`Allow to show and modify the CodaBox Connection Settings`. Add the users you want
-to this group. By default, only the :guilabel:`Administrator` is part of this group.
+#. Go to :menuselection:`Accounting --> Configuration --> Accounting: CodaBox`.
```
Indeed thx !

auva-odoo reviewed (approved):
Thanks again @RicGR98 and @chiaraprattico !
@robodoo delegate+

RicGR98 commented:
@robodoo r+
